feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonStoreProduct.kt

```diff
@@ -33,5 +85,11 @@ data class AmazonStoreProduct(
         get() = id
 }
 
+/**
+ * StoreProduct object containing Amazon-specific fields:
+ * `originalProductJson`
+ * `freeTrialPeriod`
+ * `iconUrl`
+ */
```

Comment:
Hmm I believe we won't be generating docs for anything other than the `public` and `purchases` modules. So I'm not sure if these will be used... Maybe we need to move this subclass to one of those modules, even if it's Amazon only?

Reply:
these docs will still show up if people click through in android studio though, right? i think not having them generated is an ok tradeoff until we figure out the module refactor. but open to moving it if you feel it's worth it!

Reply:
Hmm not really sure but I believe not? Currently in the docs we only generate docs for those 2 modules... We should generate the docs locally to confirm and if not, I think we should just move it to the `public` module yeah. CC @vegaro , in case you have more context. I'm not that familiar with our docs generation.

Reply:
ohh i thought our docs generation was just for creating the nice html page... i generated locally, and you're right, nothing in the amazon module is generated.

i don't know if getting docs on this (when most already exist in the higher-level `StoreProduct`) is worth moving it to public, where it might confuse non-amazon users.. i'll try out a sample project and confirm whether the javadocs are visible just navigating within android studio

Reply:
Can we still keep these docs and merge them in because I know they will be useful to me? 😇